### PR TITLE
calculate iOS shadows instead of static assignments

### LINKED ISSUE #4: Further elevation
On material guidelines: https://material.io/guidelines/material-design/elevation-shadows.html#elevation-shadows-shadows

elevation has a maximum of 24

### PR DESCRIPTION
**Pull request by JReinhold**
Fixes #4 

After a lot of trial and error and calculation, I've refactored the iOS shadow elevations to using a simple linear algorithm instead of static switch-statements.
This enables the use of an arbitrary `elevation` number instead of just 0-5.

A comparison between iOS and Android shadows resulting from the algorithm can be seen below.

It supports any elevation, however it is only tested **up to 24**, so anything exceeding that is not guaranteed to look the same across iOS and Android.
elevationcomparison

### CODE REVIEW

**Comment by alekhurst** [comment]
[edited] I'm open to having an algorithm determine the iOS elevation styles, but I think it needs a bit more tweaking. I have not tested on device yet, but in the screenshots the Android shadows are a bit lighter than the iOS shadows. 

If you're on mac, you can use Digital Color Meter to view the hex code for each pixel at a given distance from the material.

File: index.js [hunk 1]
```diff
@@ -3,16 +3,24 @@ import { View, Platform } from 'react-native';
...
     elevation: 0
-  }
+  };
 
   render() {
     const { elevation, style, ...otherProps } = this.props;
 
+	//Return default View if elevation is 0
+	if(elevation === 0){
+		return (
```
Comment (alekhurst) [inline index.js]: let's use spaces instead of tabs

**Review by alekhurst — CHANGES_REQUESTED** [review]

**Comment by JReinhold** [comment]
I've tweaked the calculations using the Digital Color Meter, and having two emulators side by side to each other. I think the shadows are very similar now, I can't imagine getting any closer.

Notice how uneven the Android elevations are at some times, especially 11 and 13 being a bit darker on iOS, while 12 is much lighter on iOS, which doesn't really make any sense.

comparev2

@alekhurst thoughts?

**Comment by alekhurst** [comment]
I'm tentative, as the shadow is drastically different for `elevation: 5` which we officially support. It looks like the values for Google's material design elevation shadows don't follow a linear equation. 

Which ios/android devices were you measuring these with, and did they have similar resolutions?

**Comment by JReinhold** [comment]
I get your point.

I haven't thought about the device's resolution having an impact, so I switched to using an iPhone 6 Plus and a Nexus 5X, which both operate at 1080x1920, and tweaked the calculations based on that.

comparev3

File: index.js [hunk 2]
```diff
@@ -3,16 +3,24 @@ import { View, Platform } from 'react-native';
...
+  };
   static defaultProps = {
     elevation: 0
-  }
+  };
 
   render() {
     const { elevation, style, ...otherProps } = this.props;
 
+    //Return default View if elevation is 0
```
Comment (alekhurst) [inline index.js]: Let's put this one below the `(Platform.OS === 'android')`, so it only effects iOS.

Also one nit: don't think we need the comment

**Review by alekhurst — DISMISSED** [review]
Sweet man, this looks great! Just one small nit

**Comment by alekhurst** [comment]
Also, it might be good to update the README:
* screenshots  (let's use the same light grey background with white squares, only need to show elevation 1-5). 
* officially support elevation values 0-24

File: README.md [hunk 3]
```diff
@@ -54,7 +54,9 @@ var styles = StyleSheet.create({
 
 ## Props
 
-- **`elevation`** _(Integer)_ - Must be between 0 and 5. The elevation you would like the elevated view to appear at.
+- **`elevation`** _(Integer)_ - Must be 0 or above, but anything higher than 24 is not officially supported.* The elevation you would like the elevated view to appear at.
```
Comment (alekhurst) [inline README.md]: nit: lets just say 

Must be between 0 and 24. The elevation you would like the elevated view to appear at.

Comment (JReinhold) [reply README.md]: done.

**Review by alekhurst — APPROVED** [review]
Sweet, go ahead and merge this puppy! I will make sure to do the npm release

**Comment by JReinhold** [comment]
perfect, good job! oh, and by the way, I don't have write access to this repo, you @alekhurst are the one that has to merge it. ;)

**Pull request merged**